src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -40,4 +43,41 @@ public boolean IsValidGameName(String gameName, Entity userEntity) {
     }
     return true;
   }
+
+  /**
+  * Creates a game entity 
+  */
+  public boolean createGameEntity(String gameName, DatastoreService datastore) {
+    
+    if(gameName == "" || datastore == null){
+        return false;
+    }
+    
+    // intializing game property values
+    Entity gameEntity = new Entity("Game");
+    ArrayList<String> userIds = new ArrayList<String>();
+    String quizQuestion = "";
+    long quiz_timestamp = 0;
+ 
+    gameEntity.setProperty("gameName", gameName);
+    gameEntity.setProperty("userIds", userIds);
+    gameEntity.setProperty("quizQuestion", quizQuestion);
+    gameEntity.setProperty("quiz_timestamp", quiz_timestamp);
+    Key gameKey = gameEntity.getKey();
+ 
+    datastore.put(gameEntity);
+
+    // getting the game id for the game id property
+    Entity entity = null;
+    try{
+      entity = datastore.get(gameKey);
+    }catch(EntityNotFoundException e){
+      return false;
```

Comment:
why return false? maybe just run an error message instead and figure out how to deal w the fact that entity is equal to null

Reply:
also I would put the key stuff below in the try block as well, since it can only run if the entity is found. otherwise u wouldnt want to put an empty entity in the datastore.

Reply:
The idea behind returning false is that if this doesn't work I'd want to break out of the function, because I don't want a game without an id to grab it with. That said, this should never happen, because what it's relying on is the entity that was created three lines above. The reason I have the try catch is because maven gets angry if I don't have it

Reply:
I used system.out to log the error; is that how you log an error? I saw some stuff about using an external logger like sentry or writing errors to a file, but I wasn't sure if I should be using either of those.

Reply:
Use `java.util.logging.Logger`
See the first Java code example here: https://cloud.google.com/appengine/docs/standard/java/logs
But you want to `log.severe` instead of `log.info`, because this is a "should never happen" error.

Reply:
Um. If it doesn't work, you may need to do something from this page: https://cloud.google.com/logging/docs/setup/java
I think you have some experience dealing with this sort of appengine config.

Reply:
The first one worked!